File: mirai-core/src/main/java/net/mamoe/mirai/message/defaults/At.java
package net.mamoe.mirai.message.defaults;

import net.mamoe.mirai.contact.QQ;
import net.mamoe.mirai.message.Message;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * At 一个人的消息.
 *
 * @author Him188moe
 */
public final class At extends Message {
    private final long target;

    public At(@NotNull QQ target) {
        this(Objects.requireNonNull(target).getNumber());
    }

    public At(long target) {
        this.target = target;
    }

    public long getTarget() {
        return target;
    }

    @Override
    public String toString() {
        // TODO: 2019/9/4 At.toString
        throw new UnsupportedOperationException();
    }
}

File: mirai-core/src/main/java/net/mamoe/mirai/message/defaults/At.kt
package net.mamoe.mirai.message.defaults

import net.mamoe.mirai.contact.QQ
import net.mamoe.mirai.message.Message
import net.mamoe.mirai.message.MessageId

/**
 * At 一个人
 *
 * @author Him188moe
 */
class At(val target: Long) : Message() {
    override val type: Int = MessageId.AT

    constructor(target: QQ) : this(target.number)

    override fun toStringImpl(): String = "[@$target]"

    override fun toByteArray(): ByteArray {
        TODO()
    }

    override fun valueEquals(another: Message): Boolean {
        if (another !is At) {
            return false
        }

        return another.target == this.target
    }
}
